build.sh

```diff
@@ -50,7 +53,7 @@ EOS
 # Maybe we can eventually parse https://static.rust-lang.org/manifests.txt
 cat <<EOS > beta
 [toolchain]
-channel = "beta"
+channel = "beta-${betaDate}"
```

Comment:
Is this valid? `rustup toolchain add beta-2023-10-03` does not work.

Reply:
Oversight on my end, will remove

Reply:
Looks like [there is a valid way](https://github.com/rust-lang/rustup/issues/1329) that actually requires a pinned date. Original idea was good, formatting was not. Will fix tomorrow.

This also seems relevant for the beta release check...

Reply:
So there is a fun issue for beta builds where the version number will include the day the last commit was made, but the archive of the day is pinned to the actual build & release date. It can be one day off, which in this case results in having to grab 2023-10-04 to get 2023-10-03...
```
❯ rustup toolchain add beta-2023-10-04
info: syncing channel updates for 'beta-2023-10-04-x86_64-pc-windows-gnu'
info: latest update on 2023-10-04, rust version 1.74.0-beta.1 (b5c050feb 2023-10-03)
info: downloading component 'cargo'
info: downloading component 'clippy'
info: downloading component 'rust-docs'
info: downloading component 'rust-mingw'
info: downloading component 'rust-std'
info: downloading component 'rustc'
info: downloading component 'rustfmt'
info: installing component 'cargo'
info: installing component 'clippy'
info: installing component 'rust-docs'
info: installing component 'rust-mingw'
info: installing component 'rust-std'
info: installing component 'rustc'
info: installing component 'rustfmt'

  beta-2023-10-04-x86_64-pc-windows-gnu installed - rustc 1.74.0-beta.1 (b5c050feb 2023-10-03)

info: checking for self-update
```

Reply:
Okay, with the structure of the manifest format, there is a `date` value before every other key, which matches the build/release date. It's currently two `curl` calls for beta, which doesn't seem _that_ horrible to me as we're now on a static 4 calls a day. My Windows installs produce at least as many on an almost daily basis when I upgrade stable-{msvc,gnu} and nightly-{msvc,gnu}.